jazz_environment-event-handler/index.js

```diff
@@ -242,64 +241,65 @@ var processItem = function (eventPayload, configData, authToken) {
 }
 
 var processEventInitialCommit = function (environmentPayload, configData, authToken) {
-	var processStgEnv = new Promise((resolve, reject) => {
-		environmentPayload.logical_id = "stg";
-		environmentPayload.status = configData.CREATE_ENVIRONMENT_STATUS;
-
-		var svcPayloadStg = {
-			uri: configData.BASE_API_URL + configData.ENVIRONMENT_API_RESOURCE,
-			method: "POST",
-			headers: { Authorization: authToken },
-			json: environmentPayload,
-			rejectUnauthorized: false
-		};
-
-		logger.info("svcPayloadStg" + JSON.stringify(svcPayloadStg));
-		request(svcPayloadStg, function (error, response, body) {
-			if (response.statusCode === 200 && body && body.data) {
-				return resolve(null, body);
-			} else {
-				logger.error("Error creating stg environment in catalog: " + JSON.stringify(response));
-				return reject({
-					"error": "Error creating stg environment for " + environmentPayload.domain + "_" + environmentPayload.service + " in  catalog",
-					"details": response.body.message
-				});
-			}
+	function processStgEnv() {
```

Comment:
There are lot of things that can be improved here. I noticed this as well. I wanted to keep the changes to source code to minimum and focus on int tests. 
- I can make the code more modular so it becomes better testable
- the code does not follow FP and that can be improved.
- usage of lodash hcould be removed.

But to avoid making too many changes and causing any regression i wanted to avoid that. But i will make the change you are suggesting. I hope you get my point above.